src/web/extractors.rs

```diff
@@ -2082,6 +2143,40 @@ mod tests {
         assert_eq!(&result.collection, "tabs");
     }
 
+    #[test]
+    fn test_quoted_bso() {
+        let payload = HawkPayload::test_default(*USER_ID);
+        lazy_static! {
+            // Some libraries encode the bso id as "{guid}" which, is technically allowed
+            // since the spec says that bso ids could be any character. However, this can
+            // screw some things up, and while we've not seen occurances of this in the
+            // wild, we're going to transcode these to "normal" GUIDs.
+            static ref ALTERED_BSO: String = format!("\"{{{}}}\"", *USER_ID);
+        }
+        let state = make_state();
+        let uri = format!(
+            "/1.5/{}/storage/tabs/{}",
+            *USER_ID,
+            urlencoding::encode(ALTERED_BSO.as_str())
+        );
+        let header = create_valid_hawk_header(&payload, &state, "GET", &uri, TEST_HOST, TEST_PORT);
+        let req = TestRequest::with_uri(&uri)
+            .data(state)
+            .header("authorization", header)
+            .header("accept", "application/json,text/plain:q=0.5")
+            .method(Method::GET)
+            .param("uid", &USER_ID_STR)
+            .param("collection", "tabs")
+            .param("bso", &ALTERED_BSO)
```

Comment:
These params are already set in the URI, so they do not need to be set here. Because the `param` calls are not needed, the lazy static for `ALTERNED_BSO` is not needed either (replace with a normal `format` call).